### PR TITLE
[image_picker] NNBD stable

### PR DESCRIPTION
**Pull request by cyanglaz**
release image_picker nnbd version as stable

### CODE REVIEW

File: packages/image_picker/image_picker/pubspec.yaml [hunk 1]
```diff
@@ -16,19 +16,18 @@ flutter:
...
 dev_dependencies:
-  video_player: ^2.0.0-nullsafety.7
   flutter_test:
     sdk: flutter
   integration_test:
     path: ../../integration_test
   mockito: ^5.0.0-nullsafety.7
-  pedantic: ^1.8.0
+  pedantic: ^1.10.0
   plugin_platform_interface: ^1.0.3
```
Comment (ditman) [inline packages/image_picker/image_picker/pubspec.yaml:29]:
```
Checking project image_picker...
  info - The library 'package:plugin_platform_interface/plugin_platform_interface.dart' is legacy, and should not be imported into a null safe library at test/image_picker_test.dart:10:8 - (import_of_legacy_library_into_null_safe)
```

Comment (ditman) [reply packages/image_picker/image_picker/pubspec.yaml:29]: This dep. needs to be set as ">=1.0.0 <3.0.0", right @stuartmorgan?

Comment (stuartmorgan) [reply packages/image_picker/image_picker/pubspec.yaml:29]: Yes

Comment (cyanglaz) [reply packages/image_picker/image_picker/pubspec.yaml:29]: this will get fixed after merging https://github.com/flutter/plugins/pull/3580

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Comment by jianhuaz** [comment]
flutter pub get
The current Dart SDK version is 2.8.4.                                  
Because showfan_cjt depends on image_picker >=0.7.0-nullsafety which requires SDK version >=2.12.0-0 <3.0.0, version solving failed.

flutter 1.22.6 dart sdk version is 2.10.5

**Comment by jianhuaz** [comment]
fatal error: module map file '/Users/***/ios/Pods/Headers/Public/flutter_plugin_android_lifecycle/flutter_plugin_android_lifecycle.modulemap' not found
    1 error generated.

flutter doctor
[✓] Flutter (Channel unknown, 1.20.4, on macOS 11.1 20C69, locale en-CN)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
[✓] Android Studio (version 4.1)
[✓] Connected device (1 available)
• No issues found!

**Comment by stuartmorgan** [comment]
Null safety support requires Flutter 1.26, which is currently on the beta channel. You should not be attempting to depend on null-safe versions of packages if you are on the stable channel.

**Pull request merged**